sympy/core/sympify.py

```diff
@@ -423,10 +423,12 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
         else:
             # float and int can coerce size-one numpy arrays to their lone
             # element.  See issue https://github.com/numpy/numpy/issues/10404.
+            from .relational import InvalidComparison
             for coerce in (float, int):
                 try:
                     return sympify(coerce(a))
-                except (TypeError, ValueError, AttributeError, SympifyError):
+                except (TypeError, ValueError, AttributeError, SympifyError,
+                        InvalidComparison):
```

Comment:
Why does this need to catch `InvalidComparison`?

Reply:
I do not recall. Quite some things have changed since I wrote the code (lots of conflicts, some quite large), so it will take some time before I have something that is close to what it was two years ago.